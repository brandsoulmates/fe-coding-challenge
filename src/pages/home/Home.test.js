import React from "react"
import { render } from "@testing-library/react"
import { act } from "react-dom/test-utils"
import axios from "axios"

import HomePage from "."

jest.mock('axios')

describe('HomePage component', () => {
  it('renders the test correctly', async () => {
    const mockData = [
      {
        id: 1,
        avatar_url: "https://avatars.githubusercontent.com/u/1?v=4",
        login: "gituserone",
      },
      {
        id: 2,
        avatar_url: "https://avatars.githubusercontent.com/u/1?v=4",
        login: "gitusertwo",
      },
      {
        id: 3,
        avatar_url: "https://avatars.githubusercontent.com/u/1?v=4",
        login: "gituserthree",
      },
    ]
    axios.get.mockResolvedValue({
      data: mockData
    })
    const result = await act( async () => render(<HomePage />))
    expect(result).toMatchSnapshot()
  })
})