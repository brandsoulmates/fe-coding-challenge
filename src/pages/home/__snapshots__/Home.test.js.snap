// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component renders the test correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="bg-sky-600"
      >
        <div
          class="mx-auto pt-5 max-w-5xl min-h-screen"
        >
          <div
            class="flex flex-wrap justify-center"
          >
            <div
              class="w-2/6"
            >
              <div
                class="m-4 px-6 pt-6 pb-10 text-center rounded-xl shadow-lg border-2 border-blue-500 bg-white"
              >
                <img
                  alt="gituserone"
                  class="mx-auto w-48 h-48 rounded-full"
                  src="https://avatars.githubusercontent.com/u/1?v=4"
                />
                <p
                  class="mt-4 mb-6 text-2xl font-medium"
                >
                  gituserone
                </p>
                <a
                  class="px-5 py-3 text-white bg-sky-700 rounded-lg"
                  href="/user/gituserone"
                >
                  VIEW PROFILE
                </a>
              </div>
            </div>
            <div
              class="w-2/6"
            >
              <div
                class="m-4 px-6 pt-6 pb-10 text-center rounded-xl shadow-lg border-2 border-blue-500 bg-white"
              >
                <img
                  alt="gitusertwo"
                  class="mx-auto w-48 h-48 rounded-full"
                  src="https://avatars.githubusercontent.com/u/1?v=4"
                />
                <p
                  class="mt-4 mb-6 text-2xl font-medium"
                >
                  gitusertwo
                </p>
                <a
                  class="px-5 py-3 text-white bg-sky-700 rounded-lg"
                  href="/user/gitusertwo"
                >
                  VIEW PROFILE
                </a>
              </div>
            </div>
            <div
              class="w-2/6"
            >
              <div
                class="m-4 px-6 pt-6 pb-10 text-center rounded-xl shadow-lg border-2 border-blue-500 bg-white"
              >
                <img
                  alt="gituserthree"
                  class="mx-auto w-48 h-48 rounded-full"
                  src="https://avatars.githubusercontent.com/u/1?v=4"
                />
                <p
                  class="mt-4 mb-6 text-2xl font-medium"
                >
                  gituserthree
                </p>
                <a
                  class="px-5 py-3 text-white bg-sky-700 rounded-lg"
                  href="/user/gituserthree"
                >
                  VIEW PROFILE
                </a>
              </div>
            </div>
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="bg-sky-600"
    >
      <div
        class="mx-auto pt-5 max-w-5xl min-h-screen"
      >
        <div
          class="flex flex-wrap justify-center"
        >
          <div
            class="w-2/6"
          >
            <div
              class="m-4 px-6 pt-6 pb-10 text-center rounded-xl shadow-lg border-2 border-blue-500 bg-white"
            >
              <img
                alt="gituserone"
                class="mx-auto w-48 h-48 rounded-full"
                src="https://avatars.githubusercontent.com/u/1?v=4"
              />
              <p
                class="mt-4 mb-6 text-2xl font-medium"
              >
                gituserone
              </p>
              <a
                class="px-5 py-3 text-white bg-sky-700 rounded-lg"
                href="/user/gituserone"
              >
                VIEW PROFILE
              </a>
            </div>
          </div>
          <div
            class="w-2/6"
          >
            <div
              class="m-4 px-6 pt-6 pb-10 text-center rounded-xl shadow-lg border-2 border-blue-500 bg-white"
            >
              <img
                alt="gitusertwo"
                class="mx-auto w-48 h-48 rounded-full"
                src="https://avatars.githubusercontent.com/u/1?v=4"
              />
              <p
                class="mt-4 mb-6 text-2xl font-medium"
              >
                gitusertwo
              </p>
              <a
                class="px-5 py-3 text-white bg-sky-700 rounded-lg"
                href="/user/gitusertwo"
              >
                VIEW PROFILE
              </a>
            </div>
          </div>
          <div
            class="w-2/6"
          >
            <div
              class="m-4 px-6 pt-6 pb-10 text-center rounded-xl shadow-lg border-2 border-blue-500 bg-white"
            >
              <img
                alt="gituserthree"
                class="mx-auto w-48 h-48 rounded-full"
                src="https://avatars.githubusercontent.com/u/1?v=4"
              />
              <p
                class="mt-4 mb-6 text-2xl font-medium"
              >
                gituserthree
              </p>
              <a
                class="px-5 py-3 text-white bg-sky-700 rounded-lg"
                href="/user/gituserthree"
              >
                VIEW PROFILE
              </a>
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
